worlds/tloz/Locations.py

```diff
@@ -0,0 +1,350 @@
+    floor_location_game_ids_early[key] = value + Rom.first_quest_dungeon_items_early
+
+floor_location_game_offsets_late = {
+    "Level 7 Item (Red Candle)": 0x4A,
+    "Level 7 Map": 0x18,
+    "Level 7 Compass": 0x5A,
+    "Level 7 Boss": 0x2A,
+    "Level 7 Triforce": 0x2B,
+    "Level 7 Key Drop (Ropes)": 0x78,
+    "Level 7 Key Drop (Goriyas)": 0x0A,
+    "Level 7 Key Drop (Stalfos)": 0x6D,
+    "Level 7 Key Drop (Moldorms)": 0x3A,
+    "Level 7 Bomb Drop (Goriyas South)": 0x69,
+    "Level 7 Bomb Drop (Keese and Spikes)": 0x68,
+    "Level 7 Bomb Drop (Moldorms South)": 0x7A,
+    "Level 7 Bomb Drop (Moldorms North)": 0x0B,
+    "Level 7 Bomb Drop (Goriyas North)": 0x1B,
+    "Level 7 Bomb Drop (Dodongos)": 0x0C,
+    "Level 7 Bomb Drop (Digdogger)": 0x6C,
+    "Level 7 Rupee Drop (Goriyas Central)": 0x38,
+    "Level 7 Rupee Drop (Dodongos)": 0x58,
+    "Level 7 Rupee Drop (Goriyas North)": 0x09,
+    "Level 8 Item (Magical Key)": 0x0F,
+    "Level 8 Item (Book of Magic)": 0x6F,
+    "Level 8 Map": 0x2E,
+    "Level 8 Compass": 0x5F,
+    "Level 8 Boss": 0x3C,
+    "Level 8 Triforce": 0x2C,
+    "Level 8 Key Drop (Darknuts West)": 0x5C,
+    "Level 8 Key Drop (Darknuts Far West)": 0x4B,
+    "Level 8 Key Drop (Pols Voice South)": 0x4C,
+    "Level 8 Key Drop (Pols Voice and Keese)": 0x5D,
+    "Level 8 Key Drop (Darknuts Central)": 0x5E,
+    "Level 8 Key Drop (Keese and Zols Entrance)": 0x7F,
+    "Level 8 Bomb Drop (Darknuts North)": 0x0E,
+    "Level 8 Bomb Drop (Darknuts East)": 0x3F,
+    "Level 8 Bomb Drop (Pols Voice North)": 0x1D,
+    "Level 8 Rupee Drop (Manhandla Entrance West)": 0x7D,
+    "Level 8 Rupee Drop (Manhandla Entrance North)": 0x4E,
+    "Level 8 Rupee Drop (Darknuts and Gibdos)": 0x6E,
+    "Level 9 Item (Silver Arrow)": 0x4F,
+    "Level 9 Item (Red Ring)": 0x00,
+    "Level 9 Map": 0x27,
+    "Level 9 Compass": 0x35,
+    "Level 9 Key Drop (Patra Southwest)": 0x61,
+    "Level 9 Key Drop (Like Likes and Zols East)": 0x56,
+    "Level 9 Key Drop (Wizzrobes and Bubbles East)": 0x47,
+    "Level 9 Key Drop (Wizzrobes East Island)": 0x57,
+    "Level 9 Bomb Drop (Blue Lanmolas)": 0x11,
+    "Level 9 Bomb Drop (Gels Lake)": 0x23,
+    "Level 9 Bomb Drop (Like Likes and Zols Corridor)": 0x25,
+    "Level 9 Bomb Drop (Patra Northeast)": 0x16,
+    "Level 9 Bomb Drop (Vires)": 0x37,
+    "Level 9 Rupee Drop (Wizzrobes West Island)": 0x40,
+    "Level 9 Rupee Drop (Red Lanmolas)": 0x12,
+    "Level 9 Rupee Drop (Keese Southwest)": 0x62,
+    "Level 9 Rupee Drop (Keese Central Island))": 0x34,
+    "Level 9 Rupee Drop (Wizzrobes Central)": 0x44,
+    "Level 9 Rupee Drop (Wizzrobes North Island)": 0x15,
+    "Level 9 Rupee Drop (Gels East)": 0x26
+}
+
+for key, value in floor_location_game_offsets_late.items():
+    floor_location_game_ids_late[key] = value + Rom.first_quest_dungeon_items_late
+
+dungeon_items = {**floor_location_game_ids_early, **floor_location_game_ids_late}
```

Comment:
do you use all 3 of these dictionaries separately? seems like it'd be better to just build a single dictionary with a single generator using `collections.ChainMap`